PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+                offeringsManager: OfferingsManager,
+                purchasesOrchestrator: PurchasesOrchestrator) {
+
+        Logger.debug(Strings.configure.debug_enabled)
+        Logger.debug(String(format: Strings.configure.sdk_version, Self.frameworkVersion))
+        Logger.user(String(format: Strings.configure.initial_app_user_id, appUserID ?? "nil appUserID"))
+
+        self.requestFetcher = requestFetcher
+        self.receiptFetcher = receiptFetcher
+        self.attributionFetcher = attributionFetcher
+        self.attributionPoster = attributionPoster
+        self.backend = backend
+        self.storeKitWrapper = storeKitWrapper
+        self.offeringsFactory = offeringsFactory
+        self.deviceCache = deviceCache
+        self.identityManager = identityManager
+        self.notificationCenter = notificationCenter
+        self.systemInfo = systemInfo
+        self.subscriberAttributesManager = subscriberAttributesManager
+        self.operationDispatcher = operationDispatcher
+        self.introEligibilityCalculator = introEligibilityCalculator
+        self.receiptParser = receiptParser
+        self.purchaserInfoManager = purchaserInfoManager
+        self.productsManager = productsManager
+        self.offeringsManager = offeringsManager
+        self.purchasesOrchestrator = purchasesOrchestrator
+
+        super.init()
+
+        identityManager.configure(appUserID: appUserID)
+        self.purchasesOrchestrator.maybeDelegate = self
+
+        systemInfo.isApplicationBackgrounded { isBackgrounded in
+            if !isBackgrounded {
+                self.operationDispatcher.dispatchOnWorkerThread {
+                    self.updateAllCaches(completion: nil)
+                }
+            } else {
+                self.purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: self.appUserID)
+            }
+        }
+
+        storeKitWrapper.delegate = purchasesOrchestrator
+        subscribeToAppStateNotifications()
+        attributionPoster.postPostponedAttributionDataIfNeeded()
+        postAppleSearchAddsAttributionCollectionIfNeeded()
+
+    }
+
+    /**
+     * Automatically collect subscriber attributes associated with the device identifiers
+     * $idfa, $idfv, $ip
+     */
+    @objc func collectDeviceIdentifiers() {
+        subscriberAttributesManager.collectDeviceIdentifiers(forAppUserID: appUserID)
+    }
+
+    deinit {
+        notificationCenter.removeObserver(self)
+        storeKitWrapper.delegate = nil
+        purchaserInfoManager.delegate = nil
+        privateDelegate = nil
+        Self.automaticAppleSearchAdsAttributionCollection = false
+        Self.proxyURL = nil
+    }
+
+    static func clearSingleton() {
+        Self.purchases = nil
+    }
+
+}
+
+extension Purchases {
+
+    static func setDefaultInstance(_ purchases: Purchases) {
+        initLock.lock()
+        if Self.purchases != nil {
+            Logger.info(Strings.configure.purchase_instance_already_set)
+        }
```

Comment:
this could also be expressed as `self.isConfigured`

Reply:
💯  ✅